python/ray/rllib/a3c/shared_model.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+import distutils.version
+from ray.rllib.a3c.policy import (
+    categorical_sample, linear,
+    normalized_columns_initializer, Policy)
+
+from ray.rllib.models.catalog import ModelCatalog
+
+class SharedModel(Policy):
+
+    def __init__(self, ob_space, ac_space, **kwargs):
+        super(SharedModel, self).__init__(ob_space, ac_space, **kwargs)
+
+    def setup_graph(self, ob_space, ac_space):
+        num_actions = ac_space.n
+        self.x = x = tf.placeholder(tf.float32, [None] + list(ob_space))
+        dist_class, dist_dim = ModelCatalog.get_action_dist(ac_space)
+        self._model = ModelCatalog.get_model(self.x, dist_dim)
```

Comment:
Note that you can pass config through after https://github.com/ray-project/ray/pull/746/files#diff-9d1cb83990f4a1213023b37591d67efbR14